app/src/main/java/com/appunite/loudius/di/NetworkModule.kt

```diff
@@ -79,6 +79,31 @@ object NetworkModule {
         }
     }
 
+    @Provides
+    @Singleton
+    @BaseAPI
+    fun provideBaseHttpClient(
+        gson: Gson,
+        @BaseAPI baseUrl: String,
+        loggingInterceptor: HttpLoggingInterceptor,
+        authInterceptor: AuthInterceptor,
+        authFailureHandler: AuthFailureHandler,
+    ): HttpClient = HttpClient(OkHttp) {
+        expectSuccess = true
+        engine {
+            addInterceptor(authInterceptor)
+            addInterceptor(TestInterceptor)
```

Comment:
question : do we always want to add TestInterceptor ? I am not a ktor expert but my six sense is telling my it may not be good idea ?

Reply:
We want to provide `TestInterceptor` because of UI tests. You can see that in `MockWebServerRule` in 63 line we're providing new url.

Reply:
I am too newbie to understand what exactly is happening there :P But this is just abit weird that we are providing TestInterceptor to something that may be used in production , are we able to provide a httpClient specfic only to tests ? I feel like we are mixing here two responsibilites : that for tests and for production stuff.